signature/policy_config_test.go

```diff
@@ -63,6 +63,40 @@ var policyFixtureContents = &Policy{
 	},
 }
 
+func TestDefaultPolicy(t *testing.T) {
+	// We can't test the actual systemDefaultPolicyPath, so override.
+	// TestDefaultPolicyPath below tests that we handle the overrides and defaults
+	// correctly.
+
+	// Success
+	policy, err := DefaultPolicy(&types.SystemContext{SignaturePolicyPath: "./fixtures/policy.json"})
+	require.NoError(t, err)
+	assert.Equal(t, policyFixtureContents, policy)
+
+	for _, path := range []string{
+		"/this/doesnt/exist", // Error reading file
+		"/dev/null",          // A failure case; most are tested in the individual method unit tests.
+	} {
+		policy, err := DefaultPolicy(&types.SystemContext{SignaturePolicyPath: path})
+		assert.Error(t, err)
+		assert.Equal(t, &Policy{Default: PolicyRequirements{NewPRReject()}}, policy)
```

Comment:
Admittedly this is all going overboard with paranoia and "callers will be using the API incorrectly".

It is easy enough, and the common thing to do, to return `nil, error`, sure. If the caller either aborts or attempt to use the `nil` value, we will fail closed (well, likely fail closed in case of the panic on `nil` dereference).

This was motivated by the worry that if there is an error, the caller may decide to default to a hard-coded policy within the calling application, and that might be “allow everything” in the worst case, or at least be not consistent with other applications. So this API is trying a default even in case of failure; the caller can detect the error and abort, or use the returned policy; and if the callers ever care about the system policy, they should _never_ override it with something hard-coded in the application.

OTOH, returning non-nil+error does not actually _prevent_ the callers from using their own default, and if they are not careful reading the API docs, that’s what they are likely to do; they would just overwrite the returned policy without even looking at it if `err != nil`. So, I guess, you are right that this is pointless…